fdbclient/FDBTypes.h

```diff
@@ -762,9 +764,18 @@ struct MappedKeyValueRef : KeyValueRef {
 
 	MappedReqAndResultRef reqAndResult;
 
+	// boundary KVs are always returned so that caller can use it as a continuation,
+	// for non-boundary KV, it is always false.
+	// for boundary KV, it is true only when the secondary query succeeds(return non-empty).
+	// Note: only MATCH_INDEX_MATCHED_ONLY and MATCH_INDEX_UNMATCHED_ONLY modes can make use of it,
+	// to decide whether the boudnary is a match/unmatch.
+	// In the case of MATCH_INDEX_ALL and MATCH_INDEX_NONE, caller should not care if boundary has a match or not.
+	bool boundaryAndExist;
```

Comment:
an exmaple: say if there are 10 index records [0 - 9], and only [0, 3, 7] have corresponding KV, say if we are in MATCHED_ONLY mode, the return would include [0,3,7,9], with 0 has `boundaryAndExist` to be `true` and 9 has `boundaryAndExist` to be `false`.

Thus, record layer can use 9 as a continuation, knowing that it is actually not a match. @jzhou77